src/components/DomainSelector.vue

```diff
@@ -16,22 +16,31 @@
   </div>
 </template>
 
-<script>
-export default {
-  name: 'domain-selector',
-  props: {
-    domainsList: Array,
-    selectedDomain: String,
-  },
-  methods: {
-    isDomainSelected(name) {
-      return this.selectedDomain === name;
-    },
-    select(newDomain) {
-      this.$emit('selectedDomain', newDomain);
-    },
-  },
-};
+<script lang="ts">
+import Vue from 'vue'
+import { Component, Prop } from 'vue-property-decorator'
+
+@Component({
+  name: 'domain-selector'
+})
+export default class DomainSelector extends Vue {
+  @Prop({
+    default: () => [],
+    type: Array
+  })
+  readonly domainsList!: Array<string>
+
+  @Prop()
+  readonly selectedDomain!: string | undefined
```

Comment:
the bang operator is not supposed to exclude `undefined` value ?

Reply:
The `!` tells the typescript compiler not to crash if the attribute ist not initialized in constructor. In case of props, I guess it's a bit trickier. Probably related to https://github.com/vuejs/vue-class-component/issues/221

Reply:
Yep, @adimascio is right: as vue will assign itself the prop depending on what the parent component bound to it, we do not pass an initiaa value

vue-class-component suggest to use the `!` to still get type hinting on props